src/bin/common.rs

```diff
@@ -13,16 +13,241 @@ use crate::grain_synth::parse_grain_table;
+  /// Compressed AV1 in IVF video output
+  #[clap(long, short, value_parser, help_heading = "INPUT/OUTPUT")]
+  pub output: PathBuf,
+  /// Overwrite output file.
+  #[clap(short = 'y', help_heading = "INPUT/OUTPUT")]
+  pub overwrite: bool,
+
+  /// Set the threadpool size
+  #[clap(long, value_parser, default_value_t = 0, help_heading = "THREADING")]
+  pub threads: usize,
+  /// Number of tile rows. Must be a power of 2. rav1e may override this based on video resolution.
+  #[clap(long, value_parser, default_value_t = 0, help_heading = "THREADING")]
+  pub tile_rows: usize,
+  /// Number of tile columns. Must be a power of 2. rav1e may override this based on video resolution.
+  #[clap(long, value_parser, default_value_t = 0, help_heading = "THREADING")]
+  pub tile_cols: usize,
+  /// Number of tiles. Tile-cols and tile-rows are overridden
+  /// so that the video has at least this many tiles.
+  #[clap(
+    long,
+    value_parser,
+    conflicts_with = "tile-rows",
+    conflicts_with = "tile-cols",
+    help_heading = "THREADING"
+  )]
+  pub tiles: Option<usize>,
+  /// Maximum number of GOPs that can be encoded in parallel
+  #[cfg(feature = "unstable")]
+  #[clap(long, value_parser, default_value_t = 0, help_heading = "THREADING")]
+  pub slots: usize,
+
+  /// Perform the first pass of a two-pass encode,
+  /// saving the pass data to the specified file for future passes
+  #[clap(
+    long,
+    value_parser,
+    value_name = "STATS_FILE",
+    help_heading = "ENCODE SETTINGS"
+  )]
+  pub first_pass: Option<PathBuf>,
+  /// Perform the second pass of a two-pass encode,
+  /// reading the pass data saved from a previous pass from the specified file
+  #[clap(
+    long,
+    value_parser,
+    value_name = "STATS_FILE",
+    conflicts_with = "first-pass",
+    help_heading = "ENCODE SETTINGS"
+  )]
+  pub second_pass: Option<PathBuf>,
+  /// Maximum number of frames to encode
+  #[clap(
+    long,
+    short,
+    value_parser,
+    default_value_t = 0,
+    help_heading = "ENCODE SETTINGS"
+  )]
+  pub limit: usize,
+  /// Skip n number of frames and encode
+  #[clap(
+    long,
+    value_parser,
+    default_value_t = 0,
+    help_heading = "ENCODE SETTINGS"
+  )]
+  pub skip: usize,
+  /// Quantizer (0-255), smaller values are higher quality [default: 100]
+  #[clap(long, value_parser, help_heading = "ENCODE SETTINGS")]
+  pub quantizer: Option<u8>,
+  /// Minimum quantizer (0-255) to use in bitrate mode [default: 0]
+  #[clap(long, value_parser, help_heading = "ENCODE SETTINGS")]
+  pub min_quantizer: Option<u8>,
+  /// Bitrate (kbps)
+  #[clap(long, short, value_parser, help_heading = "ENCODE SETTINGS")]
+  pub bitrate: Option<i32>,
+  /// Speed level (0 is best quality, 10 is fastest).
+  /// Speeds 10 and 0 are extremes and are generally not recommended.
+  #[clap(long, short, value_parser = clap::value_parser!(u8).range(0..=10), default_value_t = 6, help_heading = "ENCODE SETTINGS", long_help = build_speed_long_help())]
```

Comment:
```suggestion
  #[clap(long, short, value_parser = clap::value_parser!(u8).range(0..=10), default_value_t = 6, help_heading = "ENCODE SETTINGS", long_help = build_speed_long_help)]
```

Reply:
Interesting. It actually failed to compile for me when I omitted the `()` and succeeded without. 😕 I'll try again and see if I'm crazy or my machine is.

Reply:
Yeah it turns out it's unrelated to the parens, it's complaining because with the serialize feature on, the method returns `Option<String>` and it wants an `Option<&str>`. Which, I'm not really sure how to make it an `Option<&str>` because of lifetimes. :thinking:

Reply:
or once_cell as @Luni-4 pointed out :)

Reply:
Yep, it seems `lazy_static` is used more for this [reason](https://github.com/clap-rs/clap/issues/3828#issue-1270850697)

Reply:
Yeah, sorry, I meant to push up this change, this week has just been quite busy with IRL job. 🙂

Reply:
Should be working now